crypto/src/main/java/org/hyperledger/besu/crypto/NodeKey.java

```diff
@@ -14,16 +14,42 @@
  */
 package org.hyperledger.besu.crypto;
 
+import org.hyperledger.besu.crypto.SECP256K1.KeyPair;
 import org.hyperledger.besu.crypto.SECP256K1.PublicKey;
 import org.hyperledger.besu.crypto.SECP256K1.Signature;
 
 import org.apache.tuweni.bytes.Bytes32;
 
-public interface NodeKey {
+public class NodeKey {
 
-  Signature sign(Bytes32 dataHash);
+  private final SecurityModule crypto;
 
-  PublicKey getPublicKey();
+  public NodeKey(final SecurityModule crypto) {
+    this.crypto = crypto;
+  }
 
-  Bytes32 calculateECDHKeyAgreement(PublicKey publicKey);
+  public Signature sign(final Bytes32 dataHash) {
+    org.hyperledger.besu.crypto.Signature signature = crypto.sign(dataHash);
```

Comment:
For consistency with other files in this PR, I'd recommend the Besu classes *not* being fully qualified (e.g. `Signature`) and instead having the Tuweni classes with `SECP256K1` prefix instead.

Reply:
Agreed - reads much more nicely 👍